radixdlt/src/main/java/com/radixdlt/consensus/VertexStore.java

```diff
@@ -132,6 +123,20 @@ public Vertex commitVertex(Hash vertexId) {
 		return tipVertex;
 	}
 
+	private void storeAtom(Vertex vertexWithAtom) {
+		try {
+			this.counters.increment(CounterType.LEDGER_PROCESSED);
+			this.engine.store(vertexWithAtom.getAtom());
+			this.counters.increment(CounterType.LEDGER_STORED);
+		} catch (RadixEngineException e) {
+			// TODO: revisit TODO below since this moved from insertVertex to commitVertex
```

Comment:
Is the below TODO block still valid? In my view, it still is, though it obviously wouldn't be a vertex insertion failure anymore. @talekhinezh?

Reply:
yes, still valid. in future will need this because we still need to "soft-store" atom in engine during prepare phase.